File: lib/models/amap_road.dart
import 'package:amap_flutter_search/models/amap_geo_point.dart';
import 'package:json_annotation/json_annotation.dart';
part 'amap_road.g.dart';

@JsonSerializable()
class AMapRoad {
  ///道路ID
  String? uid;

  ///道路名称
  String? name;

  ///距离（单位：米）
  int distance = 0;

  ///方向
  String? direction;

  ///坐标点
  AMapGeoPoint? location;
  AMapRoad();
  factory AMapRoad.fromJson(Map<String, dynamic> srcJson) =>
      _$AMapRoadFromJson(srcJson);

  Map<String, dynamic> toJson() => _$AMapRoadToJson(this);
}

File: android/src/main/java/com/amap/amap_flutter_search/AMapRoad.java
package com.amap.amap_flutter_search;

public class AMapRoad {
    ///道路ID
    private String uid;

    ///道路名称
    private String name;

    ///距离（单位：米）
    int distance = 0;

    ///方向
    private String direction;

    ///坐标点
    private AMapGeoPoint location;

    public String getUid() {
        return uid;
    }

    public void setUid(String uid) {
        this.uid = uid;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getDistance() {
        return distance;
    }

    public void setDistance(int distance) {
        this.distance = distance;
    }

    public String getDirection() {
        return direction;
    }

    public void setDirection(String direction) {
        this.direction = direction;
    }

    public AMapGeoPoint getLocation() {
        return location;
    }

    public void setLocation(AMapGeoPoint location) {
        this.location = location;
    }
}
